src/app/restaurant/restaurant.component.ts

```diff
@@ -1,22 +1,32 @@
 import { Component, OnDestroy } from '@angular/core';
 import { RestaurantService } from '../services/restaurant.service';
-import { Restaurant } from '../shared/Models/restaurant';
+import { Restaurant, Review } from '../shared/Models/restaurant';
 import { ActivatedRoute } from '@angular/router';
 import { Subject, takeUntil } from 'rxjs';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-restaurant',
   templateUrl: './restaurant.component.html',
   styleUrls: ['./restaurant.component.css'],
 })
 export class RestaurantComponent implements OnDestroy {
-  restaurant: Restaurant = { name: '', description: '' };
+  reviewForm = new FormGroup({
+    rating: new FormControl<number>(0, Validators.required),
+    dateOfVisit: new FormControl('', Validators.required),
+    comment: new FormControl('', Validators.required),
+  });
+  restaurant: Restaurant = { name: '', description: '', reviews: [] };
   destroy$ = new Subject<void>();
+  stars = [1, 2, 3, 4, 5];
+  maxDate: string;
 
   constructor(
     private restaurantService: RestaurantService,
     private route: ActivatedRoute
   ) {
+    let maxDate = new Date();
+    this.maxDate = maxDate.toISOString().split('T')[0];
```

Comment:
Would this work? It would avoid having variables that aren't really necessary
```suggestion
    this.maxDate = (new Date()).toISOString().split('T')[0];
```